virtual_rainforest/models/animals/functional_group.py

```diff
@@ -28,15 +27,38 @@ class FunctionalGroup:
 
     """
 
-    def __init__(self, name: str, taxa: str, diet: str) -> None:
+    def __init__(self, name: str, taxa: str, diet: str, metabolic_type: str) -> None:
         """The constructor for the FunctionalGroup class."""
+        # Check for valid inputs
+        valid_taxa = ["mammal", "bird", "insect"]
```

Comment:
This, as well as `valid_diets` and `valid_metabolic_types` is hardcoded here, but would be better to set them as class constants, so they are easy to spot and to update if needed. Also, making them tuples, not list, as they will be inmutable. 

If they are here because this is a work in progress, that's fine, but then indicate somewhere that this will be moved, so you don't forget.

Reply:
I think diet and taxa are actually going to be short enough lists that enumerate would make sense for them as well. I went ahead and built the basics of this out but I'm curious about the match/case structure, would that be a function that replaces the dictionaries in constants.py?

Something like:
Replacing:
`ENDOTHERMIC_METABOLIC_RATE_TERMS: dict[str, tuple[float, float]] = {
    "mammal": (0.75, 0.018),
    # Mammalian metabolic rate scaling from Metabolic Ecology p210. [assumes g mass]
    "bird": (0.75, 0.05),
    # Toy values.
}`

With:
`def calculate_metabolic_rate(taxa_type: TaxaType, mass: float) -> float:
    return match taxa_type:
        case TaxaType.MAMMAL:
            return 0.75 * mass ** 0.018
        case TaxaType.BIRD:
            return 0.75 * mass ** 0.05
        case _:
            raise ValueError(f"Invalid taxa type: {taxa_type}")`

Reply:
I put together an implementation of what I think you're recommending. Let me know if I understood correctly.

Reply:
Using the dictionary with the enum as keys makes more sense.

Reply:
I got the docs issue sorted out. Can you let me know if the changes make sense?